### PR TITLE
Add branching guides

### PR DESCRIPTION
**Pull request by danieltprice**
- Point-in-time restore
https://neon-next-git-dprice-branching-docs-v2-neondatabase.vercel.app/docs/guides/branching-pitr
- Branching with the CLI
https://neon-next-git-dprice-branching-docs-v2-neondatabase.vercel.app/docs/guides/branching-neon-cli
- Branching with the API
https://neon-next-git-dprice-branching-docs-v2-neondatabase.vercel.app/docs/guides/branching-neon-api
- Branching with GitHub Actions
https://neon-next-git-dprice-branching-docs-v2-neondatabase.vercel.app/docs/guides/branching-github-actions

Simplify branching introduction topic, add links to guides.
https://neon-next-git-dprice-branching-docs-v2-neondatabase.vercel.app/docs/introduction/branching

### CODE REVIEW

**Comment by duskpoet** [comment]
I wonder if we should promote creating a branch for analytics, I think using the r/o endpoint for this matter would be easier, cause no need to clean up, and data is kept fresh.

File: content/docs/guides/branching-neon-cli.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+```bash
+export NEON_API_KEY=<neon_api_key>
+```
+
+After exporting your key, source the profile file (source `~/.bashrc` or source `~/.bash_profile`), or start a new terminal session.
+
+In your scripts or automation tooling, you can reference the environment variable like so:
+
+```bash
+neonctl branches create --api-key $NEON_API_KEY
```
Comment (duskpoet) [inline content/docs/guides/branching-neon-cli.md:89]: no need to pass it directly, NEON_API_KEY env is the default place to look for the API key

File: content/docs/guides/branching-neon-api.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+subtitle: Learn how to create and delete branches with the Neon API
+enableTableOfContents: true
+---
+The examples in this guide demonstrate creating, viewing, and deleting branches using the Neon API. For other branch-related API methods, refer to the [Neon API reference](https://api-docs.neon.tech/reference/getting-started-with-neon-api).
+
+<Admonition type="note">
+The API examples that follow may only show some of the user-configurable request body attributes that are available to you. To view all attributes for a particular method, refer to the method's request body schema in the [Neon API reference](https://api-docs.neon.tech/reference/getting-started-with-neon-api).
+</Admonition>
+
+The `jq` option specified in each example is an optional third-party tool that formats the `JSON` response, making it easier to read. For information about this utility, see [jq](https://stedolan.github.io/jq/).
```
Comment (duskpoet) [inline content/docs/guides/branching-neon-api.md:12]:
```suggestion
The `jq` program specified in each example is an optional third-party tool that formats the `JSON` response, making it easier to read. For information about this utility, see [jq](https://stedolan.github.io/jq/).
```

File: content/docs/guides/branching-githooks.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
+---
+title: Automate branching with Githooks
+subtitle: Learn how you can automatically create and delete database branches in Neon using Githooks
+enableTableOfContents: true
+isDraft: true
+---
+
+This guide leads you through creating a Githook script that automates the creation of Neon branches every time a new Git branch is created. This is possible because Neon provides an API to manage your projects, branches, and most other operations supported by the Neon Console. You can learn more about the Neon API [here](https://neon-console-docs-link.com).
```
Comment (duskpoet) [inline content/docs/guides/branching-githooks.md:8]: missed the link? :)

File: content/docs/guides/branching-githooks.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+We’ll start with a simple Githook example and incrementally build up to thr final solution. You can find the repository along with the instructions on [GitHub](https://github-link.com).
```
Comment (duskpoet) [inline content/docs/guides/branching-githooks.md:10]: here too

File: content/docs/guides/branching-githooks.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+#!/bin/bash
+if [ $3 == 0 ]; then exit; fi
+# Check if we just checked out a new branch
+if [ $3 == 1 ]; then
+  BRANCH_NAME=$(git symbolic-ref --short HEAD)
+  NUM_CHECKOUTS=`git reflog --date=local | grep -o ${BRANCH_NAME} | wc -l`
+  if [ ${NUM_CHECKOUTS} -eq 1 ]; then 
+    # Add your own PROJECT_ID and API_KEY
+    PROJECT_ID=$PROJECT_ID
+    API_KEY=$API_KEY
```
Comment (duskpoet) [inline content/docs/guides/branching-githooks.md:78]: are you sure these reassignments are needed?

File: content/docs/guides/branching-github-actions.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+```yaml
+name: Delete Neon Branch with GitHub Actions Demo
+run-name: Delete a Neon Branch 🚀
+on: [push]
+jobs:
+  delete-neon-branch:
+    uses: neondatabase/delete-branch-action@v3
+    with:
+      project_id: rapid-haze-373089
+      branch_id: br-long-forest-224191
```
Comment (duskpoet) [inline content/docs/guides/branching-github-actions.md:101]:
```suggestion
      branch: br-long-forest-224191
```

File: content/docs/guides/branching-github-actions.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+      api_key: {{ secrets.NEON_API_KEY }}
+```
+
+### Input variables
+
+- `project_id`: The ID of your Neon project. You can find this value in the Neon Console, on the **Settings** page.
+- `branch_id`: The ID of the branch you want to delete. Select **Branches** from the sidebar, and then select the branch. A branch ID has a `br-` prefix.
```
Comment (duskpoet) [inline content/docs/guides/branching-github-actions.md:108]:
```suggestion
- `branch:` The ID of the branch you want to delete. Select **Branches** from the sidebar, and then select the branch. A branch ID has a `br-` prefix.
```

it's not possible to use "branch" option to pass either branch id or branch name

File: content/docs/guides/branching-pitr.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+To create a point-in-time branch:
+
+1. Navigate to the **Branches** page in the Neon Console.
+1. Click **Create branch** to open the branch creation dialog.
+1. Enter a name for the branch. You can call it `data_recovery`, for example.
+    ![Data recovery create branch dialog](/docs/guides/data_recovery_create_branch.png)
+1. For the **Parent branch**, select the branch where the data loss occurred.
+1. Select the **Time** option to create a branch with data as it existed at a specific date and time. For example, if the data loss occurred on July 11, 2023 at 10:01am, set the time to July 11, 2023, at 10:00am, just before the faulty query was run.
+1. Leave the **Create compute endpoint** option selected. A compute endpoint is required to connect to the new branch.
+1. Click **Create Branch** to create your point-in-time branch.
```
Comment (duskpoet) [inline content/docs/guides/branching-pitr.md:24]: or you can `neonctl branches create --name data_recovery --parent 2023-06-11T10:00:00Z` :)

File: content/docs/guides/branching-githooks.md (hunk 5 above)
Comment (danieltprice) [reply content/docs/guides/branching-githooks.md:78]: I meant to remove the Githooks content from the PR. There are a few things that need to be addressed/answered before adding a Githooks guide. I'll tackle that in the future PR if we decide to add a Githooks guide.

File: content/docs/guides/branching-githooks.md (hunk 3 above)
Comment (danieltprice) [reply content/docs/guides/branching-githooks.md:8]: mean to remove this content from the PR, as per the comment below.

File: content/docs/guides/branching-neon-cli.md (hunk 1 above)
Comment (danieltprice) [reply content/docs/guides/branching-neon-cli.md:89]: Thank you. I will fix that.

File: content/docs/guides/branching-githooks.md (hunk 5 above)
Comment (duskpoet) [reply content/docs/guides/branching-githooks.md]: I also plan to give a try to this approach in the CLI code

File: content/docs/guides/branching-github-actions.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+- Using the action requires a Neon API key. For information about obtaining an API key, see [Create an API key](/docs/manage/api-keys#create-an-api-key).
+- Add your Neon API key to your GitHub Secrets:
+    1. In your GitHub repository, go to **Settings** and locate **Secrets** at the bottom of the left sidebar.
+    2. Click **Actions** > **New Repository Secret**.
+    3. Name the secret `NEON_API_KEY` and paste your API key in the **Secret** field
+    4. Click **Add Secret**.
+
+### Example
+
+The following example creates a branch from the `main` branch in your Neon project when a pull request is opened.
```
Comment (m-abdelwahab) [inline content/docs/guides/branching-github-actions.md:31]:
```suggestion
The following example creates a branch from the `main` branch in your Neon project 
```
We can either update the code snippet or remove the "when a pul request is opened" part

**Comment by danieltprice** [comment]
> I wonder if we should promote creating a branch for analytics, I think using the r/o endpoint for this matter would be easier, cause no need to clean up, and data is kept fresh.

Good point. Mahmoud mention this as well. I will drop the analytics guide and revise the content as a Read replica guide in another PR.

File: content/docs/guides/branching-github-actions.md [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+    with:
+      project_id: rapid-haze-373089
+      branch: br-long-forest-224191
+      api_key: {{ secrets.NEON_API_KEY }}
+```
+
+### Input variables
+
+- `project_id`: The ID of your Neon project. You can find this value in the Neon Console, on the **Settings** page.
+- `branch:` The ID of the branch you want to delete. Select **Branches** from the sidebar, and then select the branch. A branch ID has a `br-` prefix.
```
Comment (duskpoet) [inline content/docs/guides/branching-github-actions.md:108]:
```suggestion
- `branch:` The ID or name of the branch you want to delete. Select **Branches** from the sidebar, and then select the branch. A branch ID has a `br-` prefix.
```

**Review by duskpoet — APPROVED** [review]

**Review by m-abdelwahab — DISMISSED** [review]

**Review by m-abdelwahab — APPROVED** [review]

**Review by m-abdelwahab — APPROVED** [review]

**Pull request merged**